odxtools/parameters/codedconstparameter.py

```diff
@@ -15,15 +16,15 @@
     from ..diaglayer import DiagLayer
 
 
+@dataclass
 class CodedConstParameter(Parameter):
 
-    def __init__(self, *, diag_coded_type: DiagCodedType, coded_value: AtomicOdxType, **kwargs):
-        super().__init__(parameter_type="CODED-CONST", **kwargs)
+    diag_coded_type: DiagCodedType
+    coded_value: AtomicOdxType
 
-        self.diag_coded_type = diag_coded_type
-        if not isinstance(coded_value, (int, bytes, bytearray)):
-            odxraise()
-        self.coded_value = coded_value
+    @property
+    def parameter_type(self) -> str:
```

Comment:
TIP:

Consider using a `Literal` typehint for return value

Reply:
good idea. implemented, thanks: 27a43512cd58203aa44d55aabaa2dded1e4cfd49